pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/TenantResources.java

```diff
@@ -110,6 +112,62 @@ public void createTenant(String tenantName, TenantInfo ti) throws MetadataStoreE
         return namespaces;
     }
 
+    public CompletableFuture<List<String>> getListOfNamespacesAsync(String tenant) {
+        CompletableFuture<List<String>> result = new CompletableFuture<>();
```

Comment:
Could we try to use ``thenAccept`` or  ``thenCompose`` to refactor it?

Reply:
Good catch!